src/main/java/org/folio/service/mapping/settings/MappingSettingsProvider.java

```diff
@@ -0,0 +1,45 @@
+package org.folio.service.mapping.settings;
+
+import io.vertx.core.json.JsonObject;
+import org.folio.clients.InventoryClient;
+import org.folio.service.mapping.processor.translations.Settings;
+import org.folio.util.OkapiConnectionParams;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+/**
+ * The component is responsible to provide settings and parameters needed for mapping.
+ * Caches settings for the given job id.
+ * If requested settings are not found by the given job id then performs loading.
+ */
+@Component
+public class MappingSettingsProvider {
+  private static final int CACHE_EXPIRATION_AFTER_ACCESS_SECONDS = 60;
```

Comment:
Currently, for a medium sized load the job is running for 2-3 hours. Given that I think we can just do cache eviction when the job has ended right? because if it is  a minute for medium sized jobs we are still fetching all the entries per read

Reply:
The entries do not get expired every 60 seconds, because no one chunk of uuids does not take more than 60 seconds of processing.
It is about to expire entries after the last access, that will be 2-3 hours + 60 seconds in this case

Reply:
I got that point that it expires after 60 seconds of the last access, it doesn't work if the UUIDs are intertwined , say 100 entries of records in SRS followed by 100 entries to generate records on the fly. 
Anyway not a huge deal , but let's try to keep an eye on  cache misses if required in the future.